cpp/include/raft/cudart_utils.h

```diff
@@ -97,27 +97,96 @@ struct cuda_error : public raft::exception {
 
 namespace raft {
 
-/** helper method to get max usable shared mem per block parameter */
-inline int get_shared_memory_per_block() {
-  int dev_id;
-  CUDA_CHECK(cudaGetDevice(&dev_id));
-  int smem_per_blk;
-  CUDA_CHECK(cudaDeviceGetAttribute(
-    &smem_per_blk, cudaDevAttrMaxSharedMemoryPerBlock, dev_id));
-  return smem_per_blk;
-}
-/** helper method to get multi-processor count parameter */
-inline int get_multi_processor_count() {
-  int dev_id;
-  CUDA_CHECK(cudaGetDevice(&dev_id));
-  int mp_count;
-  CUDA_CHECK(
-    cudaDeviceGetAttribute(&mp_count, cudaDevAttrMultiProcessorCount, dev_id));
-  return mp_count;
+/** Helper method to get to know warp size in device code */
+__host__ __device__ constexpr inline int warp_size() { return 32; }
+
+__host__ __device__ constexpr inline unsigned int warp_full_mask() {
+  return 0xffffffff;
 }
 
-/** Helper method to get to know warp size in device code */
-constexpr inline int warp_size() { return 32; }
+/**
+ * @brief A kernel grid configuration construction gadget for simple one-dimensional mapping
+ * elements to threads.
+ */
+class grid_1d_thread_t {
+ public:
```

Comment:
Perhaps better use an aggregate (`struct`) rather than a `class` with `public:` data members.

Reply:
Yeah... this blindly follows cuDF (https://github.com/rapidsai/cudf/blob/branch-0.15/cpp/include/cudf/detail/utilities/cuda.cuh#L40); cuDF is using class for this.

I guess struct may make slightly more sense but in practice, the difference is pretty minimal (and this has a custom constructor as well... though nothing prevents struct from having a constructor), so I put little more weight on consistency across RAPIDS (this maybe helpful when we embrace RAPIDS-core).

But I am fine with either struct or class... so what do you think, or any other thoughts from anyone else? I don't have a strong opinion on this, so someone has a strong one, I can embrace it.